src/irep2/irep2.h

```diff
@@ -298,11 +285,15 @@ typedef irep_container<expr2t> expr2tc;
 typedef std::pair<std::string, std::string> member_entryt;
 typedef std::list<member_entryt> list_of_memberst;
 
+class irep2t : public std::enable_shared_from_this<irep2t>
+{
+};
+
```

Comment:
wouldn't we allow implicit conversion by setting the same base class for type2t and expr2t? If so, we should remove this base class

Reply:
The purpose is to allow implicit conversion of any type2t or expr2t to irep2t. Can't implicitly cast down from irep2t to the others, though. I don't need this base class in this PR and can remove it, however, for addressing #1208 I would eventually like to be able to build the explicit graph with both type2t and expr2t nodes inside it.